build/build.ps1

```diff
@@ -0,0 +1,256 @@
+    # Use Invoke-Expression so that $dotnetInstallVerbosity is not positionally bound when empty
+    Invoke-Expression -Command "$dotnetInstallScript -Version $dotnetCliVersion $dotnetInstallVerbosity"
+
+    if($LASTEXITCODE -ne 0) {
+      throw "Failed to install stage0"
+    }
+  }
+
+  # Install 1.0 shared framework
+  $netcoreApp10Version = "1.0.5"
+  $netCoreApp10Dir = Join-Path $env:DOTNET_INSTALL_DIR "shared\Microsoft.NETCore.App\$netcoreApp10Version"
+
+  if (!(Test-Path $netCoreApp10Dir)) {
+    # Use Invoke-Expression so that $dotnetInstallVerbosity is not positionally bound when empty
+    Invoke-Expression -Command "$dotnetInstallScript -Channel `"Preview`" -Version $netcoreApp10Version -SharedRuntime $dotnetInstallVerbosity"
+
+    if($LASTEXITCODE -ne 0) {
+      throw "Failed to install stage0"
+    }
+  }
+
+  # Install 1.1 shared framework
+  $netcoreApp11Version = "1.1.2"
+  $netcoreApp11Dir = Join-Path $env:DOTNET_INSTALL_DIR "shared\Microsoft.NETCore.App\$netcoreApp11Version"
+
+  if (!(Test-Path $netCoreApp11Dir)) {
+    # Use Invoke-Expression so that $dotnetInstallVerbosity is not positionally bound when empty
+    Invoke-Expression -Command "$dotnetInstallScript -Channel `"Release/1.1.0`" -Version $netcoreApp11Version -SharedRuntime $dotnetInstallVerbosity"
+
+    if($LASTEXITCODE -ne 0) {
+      throw "Failed to install stage0"
+    }
+  }
+
+  # Put the stage 0 on the path
+  $env:PATH = "$env:DOTNET_INSTALL_DIR;$env:PATH"
+
+  # Disable first run since we want to control all package sources
+  $env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+
+  # Don't resolve runtime, shared framework, or SDK from other locations
+  $env:DOTNET_MULTILEVEL_LOOKUP=0
+}
+
+function InstallNuGet {
+  $nugetInstallDir = Join-Path $RepoRoot "artifacts\.nuget"
+  $nugetExe = Join-Path $nugetInstallDir "nuget.exe"
+
+  if (!(Test-Path -path $nugetExe)) {
+    Create-Directory $nugetInstallDir
+    Invoke-WebRequest "https://dist.nuget.org/win-x86-commandline/latest/nuget.exe" -UseBasicParsing -OutFile $nugetExe
+  }
+}
+
+function LocateVisualStudio {
+  $vswhereVersion = GetVSWhereVersion
+  $vsWhereDir = Join-Path $ToolsRoot "vswhere\$vswhereVersion"
+  $vsWhereExe = Join-Path $vsWhereDir "vswhere.exe"
+
+  if (!(Test-Path $vsWhereExe)) {
+    Create-Directory $vsWhereDir
+    Invoke-WebRequest "http://github.com/Microsoft/vswhere/releases/download/$vswhereVersion/vswhere.exe" -UseBasicParsing -OutFile $vswhereExe
+  }
+
+  $vsInstallDir = & $vsWhereExe -latest -property installationPath -requires Microsoft.Component.MSBuild -requires Microsoft.VisualStudio.Component.VSSDK -requires Microsoft.Net.Component.4.6.TargetingPack -requires Microsoft.VisualStudio.Component.Roslyn.Compiler -requires Microsoft.VisualStudio.Component.VSSDK
+
+  if (!(Test-Path $vsInstallDir)) {
+    throw "Failed to locate Visual Studio (exit code '$lastExitCode')."
+  }
+
+  return $vsInstallDir
+}
+
+function Build {
+  InstallDotNetCli
+  InstallNuget
```

Comment:
I think you want to do this only if $restore is true.

Reply:
Its used by tests